src/process_user_code.py

```diff
@@ -43,16 +46,39 @@ def run(self):
             read_val = sys.stdin.readline()
             sys.stdin.flush()
             try:
-                new_state = json.loads(read_val)
-                for event in CONSTANTS.EXPECTED_INPUT_EVENTS_CPX:
-                    cpx._Express__state[event] = new_state.get(
-                        event, cpx._Express__state[event]
-                    )
+                new_state_message = json.loads(read_val)
+                device = new_state_message.get(CONSTANTS.ACTIVE_DEVICE_FIELD)
+                new_state = new_state_message.get(CONSTANTS.STATE_FIELD, {})
+
+                if device == CPX:
+                    update_cpx(new_state)
+                elif device == MICROBIT:
+                    update_microbit(new_state)
+                else:
+                    raise Exception(CONSTANTS.DEVICE_NOT_IMPLEMENTED_ERROR)
 
             except Exception as e:
                 print(CONSTANTS.ERROR_SENDING_EVENT, e, file=sys.stderr, flush=True)
 
 
+def update_cpx(new_state):
+    for event in CONSTANTS.EXPECTED_INPUT_EVENTS_CPX:
+        cpx._Express__state[event] = new_state.get(event, cpx._Express__state[event])
+
+
+def update_microbit(new_state):
+    for button in CONSTANTS.EXPECTED_INPUT_BUTTONS_MICROBIT:
+        previous_pressed = None
+        exec(f"previous_pressed = mb.{button}.get_presses()")
```

Comment:
I think you meant to use `is_pressed()` here?
`get_presses()` resets the "button clicked" counter and we probably want to refrain from using it in the backend.
From the method docstring:
```
Returns the running total of button presses, and resets this total
to zero before returning.
```

Reply:
oops, called the wrong function! thanks for catching that!